google/cloud/alloydb/connector/instance.py

```diff
@@ -33,6 +34,29 @@
 
 logger = logging.getLogger(name=__name__)
 
+INSTANCE_URI_REGEX = re.compile(
+    "projects/([^:]+(:[^:]+)?)/locations/([^:]+)/clusters/([^:]+)/instances/([^:]+)"
+)
+
+
+def parse_instance_uri(instance_uri: str) -> Tuple[str, str, str, str]:
+    # should take form "projects/<PROJECT>/locations/<REGION>/clusters/<CLUSTER>/instances/<INSTANCE>"
+    if INSTANCE_URI_REGEX.fullmatch(instance_uri) is None:
+        raise ValueError(
+            "Arg `instance_uri` must have "
+            "format: projects/<PROJECT>/locations/<REGION>/clusters/<CLUSTER>/instances/<INSTANCE>, projects/<DOMAIN>:<PROJECT>/locations/<REGION>/clusters/<CLUSTER>/instances/<INSTANCE>"
+            f"got {instance_uri}."
+        )
+
+    instance_uri_split = INSTANCE_URI_REGEX.split(instance_uri)
+
+    return (
```

Comment:
nit: let's remove blank lines
```suggestion
        )
    instance_uri_split = INSTANCE_URI_REGEX.split(instance_uri)
    return (
```

Reply:
Hi @jackwotherspoon, I have removed the empty lines.